python/tvm/runtime/profiling.py

```diff
@@ -0,0 +1,57 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Runtime profiling functions."""
+import tvm._ffi
+
+
+def start_timer(ctx):
+    """
+    Start a low-overhead device specific timer.
+
+    Params
+    ------
+    ctx: TVMContext
+        The context to get a timer for.
+
+    Returns
+    -------
+    timer_start: function
+        A function that stops the device specific timer. Calling this function
+        stops the timer and returns another function that returns the elapsed
+        time in nanoseconds. This third function should be called as late as
```

Comment:
explain the data type of the return value. if it is floating point, why not seconds?

Reply:
also, what will happen if a wraparound occurs? it would be great to describe it in this api doc

Reply:
I doubt wrap around will occur. You need to time something for longer than 584.6 years.

Reply:
what happens if underflow occurs because the system time changed?

Reply:
You get a negative duration (int64_t is signed). Right now I'm using high_resolution_clock which maybe not be monotonic. We could switch to steady clock, but it maybe not be as high resolution.